### PR TITLE
Donation System with Rewards and Reskins

### PR DESCRIPTION
**Pull request by ZephyrTFA**
donation system, that I said I would work on a long time ago.
But hey, it's here at last, right?

its currently completely loaded from json files, but eventually I will be adding db support and deprecating the json files

:cl:
add: Donator Rewards, by default everyone can get a free COOKIE!!!
:cl:/

### CODE REVIEW

**Comment by MarkSuckerberg** [comment]
Can't say I'm fully certain this is something we want, always wary of any form of monetisation.

**Comment by ZephyrTFA** [comment]
The intended purpose of this is for donators to get reskins of items and/or vanity items.
I'm fine with this so long as we never add anything actually gamer into them like a certain russian downstream

File: code/modules/donator/_donator.dm [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+
+	/// typecache of eligible rewards for this donator
+	var/list/flat_rewards = list(
+		/obj/item/reagent_containers/food/snacks/cookie = TRUE
+	)
+
+	/// list of conversion rewards for this donator
+	/// Expected format: base type -> list of convertible types
+	var/list/conversion_rewards = list(
+		/obj/item/screwdriver = list(/obj/item/screwdriver/old, /obj/item/screwdriver/caravan),
```
Comment (ZephyrTFA) [inline code/modules/donator/_donator.dm]:
```suggestion
```

File: code/modules/donator/_donator.dm [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
+	)
+
+	/// list of reskin rewards for this donator
+	/// Should be an assosciative list indexed by type with a value which is a list of skins
+	var/list/reskin_rewards = list(
+		/obj/item/screwdriver = list("fakeiconstate", "alsonotreal")
```
Comment (ZephyrTFA) [inline code/modules/donator/_donator.dm]:
```suggestion
```

**Comment by ZephyrTFA** [comment]
I'm just going to leave one more comment.
DO NOT USE THIS FOR MECHANICAL REWARDS.
The walls have ears...

**Pull request merged**